spec/lib/sso_cookie_jar_spec.rb

```diff
@@ -2,42 +2,92 @@
 
 RSpec.describe SsoCookieJar do
 
-  let(:mock_request) {
+  let(:json_serializer) {
+    ActionDispatch::Cookies::JsonSerializer
+  }
+
+  let(:sso_shared_secret) {
+    Rails.application.secrets.sso[:shared_secret]
+  }
+
+  let(:sso_shared_salt) {
+    Rails.application.secrets.sso[:shared_secret_salt] # || 'cookie'
+  }
+
+  let(:key_generator) {
+    ActiveSupport::CachingKeyGenerator.new(
+      ActiveSupport::KeyGenerator.new(sso_shared_secret, iterations: 1000)
+    )
+  }
+
+  let(:request) {
     OpenStruct.new(
```

Comment:
Maybe you can use a TestRequest.

And you shouldn't set all these headers IMO. You should use something similar to a normal request.

Reply:
Wow. It just worked 😄 . `let(:request)    { ActionController::TestRequest.create(:test) }`.
Fixed.